GitUI/CommandsDialogs/ScriptOptionsProvider.cs

```diff
@@ -0,0 +1,49 @@
+using GitCommands;
+using GitUI.ScriptsEngine;
+
+namespace GitUI.CommandsDialogs;
+
+internal class ScriptOptionsProvider : IScriptOptionsProvider
+{
+    private const string _selectedFiles = "SelectedFiles";
+    private const string _lineNumber = "LineNumber";
+
+    private Func<IEnumerable<string>?> _getSelectedFiles;
+    private Func<IFullPathResolver> _getFullPathResolver;
+    private Func<int?> _getCurrentLineNumber;
+
+    public ScriptOptionsProvider(Func<IEnumerable<string>?> getSelectedFiles, Func<IFullPathResolver> getFullPathResolver, Func<int?> getCurrentLineNumber)
+    {
+        _getSelectedFiles = getSelectedFiles;
+        _getFullPathResolver = getFullPathResolver;
+        _getCurrentLineNumber = getCurrentLineNumber;
+    }
+
+    public ScriptOptionsProvider(FileStatusList? fileStatusList, Func<IFullPathResolver> getFullPathResolver, Func<int?> getCurrentLineNumber)
+        : this(() => fileStatusList?.SelectedItems?.Select(item => item.Item.Name), getFullPathResolver, getCurrentLineNumber)
```

Comment:
```suggestion
    public ScriptOptionsProvider(FileStatusList fileStatusList, Func<IFullPathResolver> getFullPathResolver, Func<int?> getCurrentLineNumber)
        : this(() => fileStatusList.SelectedItems.Select(item => item.Item.Name), getFullPathResolver, getCurrentLineNumber)
```

Reply:
The issue is that the `FileStatusList` _can_ be null. Since it has to receive `FileStatusList? _currentFilesList` from `FormCommit`. It is initially `null` until one of the lists is selected. `ScriptOptionsProvider` has to handle this situation gracefully, so it _must_ accept a nullable.

Reply:
I think `Validates.NotNull(_currentFilesList)` would work. There must be selection if a script is run on a file.
Though there is a simpler and appropriate solution: use `null` instead of creating `ScriptOptionsProvider` which does not replace anything.

Reply:
It's been split to a separate PR, but in the presence of `{if:SelectedFiles}`, it is possible to run the script even if no file is selected. In any case, this possibility is mandatory, for the case of running the script from a context that offers no files, like the revision list.

However, passing `null` for the `IScriptOptionsProvider` is not perfectly equivalent here, because it applies not to one option, but to all of them. In this case there are two, "SelectedFiles" and "LineNumber". Moreover, it causes `IScriptOptionsProviderOptions` to be empty as well. In other words, it obscures the difference between "One option is not available currently" and "No option is available in this context". These two things are not the same.